python/dask_cudf/dask_cudf/backends.py

```diff
@@ -37,20 +37,36 @@ def concat_cudf(
 
 try:
 
-    from dask.dataframe.methods import group_split, hash_df
-
-    @hash_df.register(cudf.DataFrame)
-    def hash_df_cudf(dfs):
-        return dfs.hash_columns()
-
-    @hash_df.register(cudf.Index)
-    def hash_df_cudf_index(ind):
-        from cudf.core.column import column, numerical
-
-        cols = [column.as_column(ind)]
-        return cudf.Series(numerical.column_hash_values(*cols))
-
-    @group_split.register(cudf.DataFrame)
+    from dask.dataframe.utils import group_split_dispatch, hash_object_dispatch
+    from cudf.core.column import column, CategoricalColumn, StringColumn
+
+    def _string_safe_hash(df):
+        frame = df.copy(deep=False)
+        for col in frame.columns:
+            if isinstance(frame[col]._column, StringColumn):
+                frame[col] = frame[col]._column.as_numerical_column("int32")
```

Comment:
Does this guarantee the same value for arbitrary strings?

Reply:
> Does this guarantee the same value for arbitrary strings?

Thanks for raising this question - No, it seems that `as_numerical_column` does not work the way I originally thought.  I will push a fix for this soon.

Reply:
I believe there's a specific NVStrings hashing method that can be used here for the time being. libcudf++ will address this much more cleanly.

Reply:
Yes - That is exactly the solution I am refering to (I believe) - Just dealing with some local environment issues at the moment :)

Reply:
The current approach (just pushed) should guarantee the same value for arbitrary strings.  Here is a simple test case where the numerical hash for "fish" matches in two separate dataframes:

```
In [1]: import cudf 
   ...: import rmm 
   ...: import cudf._lib as libcudf 
   ...: from cudf.core.buffer import Buffer                                                                                                                                                                     

In [2]: df1 = cudf.DataFrame({"a": ["cat", "dog", "fish"]}) 
   ...: out_dev_arr_1 = rmm.device_array(len(df1), dtype="int32") 
   ...: ptr_1 = libcudf.cudf.get_ctype_ptr(out_dev_arr_1) 
   ...: df1["a"]._column.data.hash(devptr=ptr_1) 
   ...: df1["b"] = cudf.Series(Buffer(out_dev_arr_1))                                                                                                                                                           

In [3]: df2 = cudf.DataFrame({"a": ["fish", "bird", "ant"]}) 
   ...: out_dev_arr_2 = rmm.device_array(len(df2), dtype="int32") 
   ...: ptr_2 = libcudf.cudf.get_ctype_ptr(out_dev_arr_2) 
   ...: df2["a"]._column.data.hash(devptr=ptr_2) 
   ...: df2["b"] = cudf.Series(Buffer(out_dev_arr_2))                                                                                                                                                           

In [4]: df1                                                                                                                                                                                                     
Out[4]: 
      a           b
0   cat -1033718724
1   dog  1317114013
2  fish  1263633836

In [5]: df2                                                                                                                                                                                                     
Out[5]: 
      a           b
0  fish  1263633836
1  bird  -722947564
2   ant  1769099976
```